spec/codebreaker/game_spec.rb

```diff
@@ -0,0 +1,71 @@
+require 'spec_helper'
+
+module Codebreaker
+  RSpec.describe Game do
+    context '#initialize' do
+      let(:game) { Game.new }
+      before do
+        game.start
+      end
+
+      it 'saves secret code' do
+        expect(game.instance_variable_get(:@secret_code)).not_to be_empty
+      end
+
+      it 'saves 4 numbers secret code' do
+        expect(game.instance_variable_get(:@secret_code).size).to eq 4
+      end
+
+      it 'saves secret code with numbers from 1 to 6' do
+        expect(game.instance_variable_get(:@secret_code).to_s).to match(/[1-6]+/)
+      end
+
+      it 'max attempt set to 8' do
+        expect(game.instance_variable_get(:@max_attempts)).to eq 8
```

Comment:
magical number, use constant instead so the test will not fail when you change attempts in your code